### PR TITLE
Add module to sample, validate and evaluate goals for a "move cube" task

### PR DESCRIPTION
**Pull request by luator**
## Description

Add a `tasks` subpackage with one module for the "move cube" task.

The module provides functions for sampling, validating and evaluating goal poses for the cube.  The idea is that for following tasks an equivalent module with the same functions is added.

## Open Questions
- [x] Should we better create a separate package for this (e.g. "finger_tasks")?
- [x] Should there be a minimum distance to the current position when sampling a goal (to avoid very easy goals)?
- [ ] What should be the maximum height?
- [x] Should there be a minimum height for level 2?
- [x] Does the current cost function make sense?

## How I Tested

Not much yet.  I should add unit tests but would first like to see if the intended behaviour is good.

## I fulfilled the following requirements

[//]: # "Please make sure you followed these steps before requesting a review."
[//]: # "Check the boxes in the list below, when done."

- [x] All new code is formatted according to our style guide (for C++ run clang-format, for Python, run flake8 and fix all warnings).
- [x] All new functions/classes are documented and existing documentation is updated according to changes.
- [x] No commented code from testing/debugging is kept (unless there is a good reason to keep it).

### CODE REVIEW

File: python/robot_fingers/tasks/move_cube.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        raise ValueError("Invalid difficulty %d" % difficulty)
+
+    position = np.array((x, y, z))
+
+    # random yaw angle (relative to current_orientation if given)
+    yaw = random.uniform(0, 2 * np.pi)
+
+    orientation = Rotation.from_euler("z", yaw).as_quat()
+    if current_orientation is not None:
+        orientation = orientation * current_orientation
```
Comment (mwuethri) [inline python/robot_fingers/tasks/move_cube.py:108]: i guess this does actually proper application of a rotation (i.e. not just elment-wise multiplication)? is this also guaranteed if the current_orientation is simply a numpy array (representing a quaternion)? just to make sure that there is no potential for error here

File: python/robot_fingers/tasks/move_cube.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        Cost of the actual pose w.r.t. to the goal pose.  Lower value means
+        that the actual pose is closer to the goal.  Zero if actual == goal.
+    """
+    # Use DISP distance (max. displacement of the corners)
+    goal_corners = get_cube_corner_positions(goal_position, goal_orientation)
+    actual_corners = get_cube_corner_positions(
+        actual_position, actual_orientation
+    )
+
+    disp = max(np.linalg.norm(goal_corners - actual_corners, axis=1))
```
Comment (mwuethri) [inline python/robot_fingers/tasks/move_cube.py:164]: very elegant solution. i wonder, is the max the right thing here? or the mean? 

could this lead to counterintuitive behavior?

Comment (luator) [reply python/robot_fingers/tasks/move_cube.py:164]: This is the "displacement" metric as defined, for example, [here (p. 190, Example 5.6)](http://lavalle.pl/planning/bookbig.pdf).

Intuitively I would think that max is better than mean in cases of rotations (e.g. when the cube is rotated around one edge, such that the corners of this edge are still at their original position).  But this is just my gut feeling, I'm not aware of a systematic comparison.

File: python/robot_fingers/tasks/move_cube.py (hunk 1 above)
Comment (luator) [reply python/robot_fingers/tasks/move_cube.py:108]: Good point, I missed that.  Yes, the intention is to combine the two rotations but for this `current_orientation` needs to be converted to a `Rotation` object first (this part is not properly tested yet).

Comment (luator) [reply python/robot_fingers/tasks/move_cube.py]: Fixed.

**Comment by luator** [comment]
Moved this to the pybullet_fingers package instead of adding it here.

**Pull request closed without merging**